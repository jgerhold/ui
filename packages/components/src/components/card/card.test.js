import { mount } from '@vue/test-utils'
import Card from './Card'

describe('Card', () => {
  describe('component', () => {
    test('should be a Vue instance', () => {
      const wrapper = mount(Card)
      expect(wrapper.isVueInstance()).toBeTruthy()
    })

    test('should provide defaults', async () => {
      const wrapper = await mount(Card, {
        slots: {
          default: '<span>Card Inner</span>'
        }
      })

      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('props', () => {
    const props = [
      {
        prop: 'headerColor',
        value: '#fff'
      },
      {
        prop: 'backgroundColor',
        value: '#fff'
      }
    ]

    props.forEach(({ prop, value }) => {
      test(`should have a property for ${prop}`, async () => {
        const wrapper = await mount(Card, {
          slots: {
            default: '<span>Card Inner</span>'
          },
          propsData: {
            [prop]: value
          }
        })

        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
