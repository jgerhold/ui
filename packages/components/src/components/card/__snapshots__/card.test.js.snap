// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should provide defaults 1`] = `
<div
  class="card"
  style="background: rgb(255, 255, 255);"
>
  <div
    class="card-header"
    style="background: rgb(43, 138, 198);"
  />
   
  <div
    class="card-body"
  >
    <span>
      Card Inner
    </span>
  </div>
</div>
`;

exports[`Card props should have a property for backgroundColor 1`] = `
<div
  class="card"
  style="background: rgb(255, 255, 255);"
>
  <div
    class="card-header"
    style="background: rgb(43, 138, 198);"
  />
   
  <div
    class="card-body"
  >
    <span>
      Card Inner
    </span>
  </div>
</div>
`;

exports[`Card props should have a property for headerColor 1`] = `
<div
  class="card"
  style="background: rgb(255, 255, 255);"
>
  <div
    class="card-header"
    style="background: rgb(255, 255, 255);"
  />
   
  <div
    class="card-body"
  >
    <span>
      Card Inner
    </span>
  </div>
</div>
`;
